tools/ddrgen/src/scanners/dwarf/AixDwarfParser.cpp

```diff
@@ -705,18 +279,15 @@ parseDeclarationLine (const string data, Parsed_Declaration_Data *parsedData)
 static int
 parseDwarfDie(const string line)
 {
-	
 	int ret = DW_DLV_OK;
 	/* get the type of the data of the line being parsed */
 	Parsed_Declaration_Data *parsedDeclarationData = new Parsed_Declaration_Data();
```

Comment:
1. I was not trying to make a point about how to initialize it. It was an example. Temporary local variables like this should not use new to allocate memory. Its unnecessary, adds excess error checking, and adds complication with freeing the memory. As is, in the case of running out of memory, your new call will return null and the program will crash when it attempts to use the variable. No null check exists in the code. Don't add null checks, just get rid of this use of new and of Parsed_Declaration_Data.

2. The Parsed_Declaration_Data structure should not even exist in this commit anyway, since the point was to populate the Dwarf structures instead of it.

3. When you do need to use new, it should be paired with delete() and not free(). free() is for malloc()'ed memory only. It also requires error checking: you should always check if the result of new/malloc is null before using it and be sure to only free variables that are not null.